command_line_assistant/utils/cli.py

```diff
@@ -214,3 +235,141 @@ def create_subparser(parser: SubParsersAction, name: str, help: str) -> Argument
     )
 
     return custom_parser
+
+
+def command(
+    name: str, help: Optional[str] = None, description: Optional[str] = None
+) -> Callable[[CommandFunc], CommandFunc]:
+    """Decorator to register a command function.
+
+    Args:
+        name (str): Command name
+        help (Optional[str], optional): Short help text
+        description (Optional[str], optional): Longer description
+
+    Returns:
+        Callable[[CommandFunc], CommandFunc]: Decorator function
+    """
+
+    def decorator(func: CommandFunc) -> CommandFunc:
+        """Decorator to register a command function.
+
+        Args:
+            func (CommandFunc): Command function
+
+        Returns:
+            CommandFunc: Command function
+        """
+        cmd = Command(name, func, help, description)
+        _commands[name] = cmd
+
+        logger.debug("Registered command: %s", name)
+        return func
+
+    return decorator
+
+
+def argument(*args: str, **kwargs: Any) -> Callable[[CommandFunc], CommandFunc]:
```

Comment:
the length of the comment blocks make this difficult to read. I didn't see this was returning a function at first.